pkg/caveats/types/ipaddress.go

```diff
@@ -26,7 +26,7 @@ func MustParseIPAddress(ip string) IPAddress {
 	return ipAddress
 }
 
-var ipaddressCelType = types.NewTypeValue("IPAddress", traits.ReceiverType)
+var ipaddressCelType = cel.ObjectType("IPAddress", traits.ReceiverType)
```

Comment:
Does the `IPAddress` type have fields that are accessible, or just methods? If you don't expose any fields to the caller directly, you can consider making this an `OpaqueType`.

Reply:
Just the `in_cidr` function right now. You'd recommend making it Opaque for that?

Reply:
Either way is fine. If you mark something as an object type, then there's some expectation that it will have fields that are directly accessible in the expression. If you always intend to expose functionality through methods, then consider an `OpaqueType` since it's slightly less overhead.

Reply:
Gotcha. Changed